master/docs/manual/cfg-buildsteps.rst

```diff
@@ -1240,8 +1267,65 @@ The arguments are identical to :bb:step:`ShellCommand`.
 ::
 
     from buildbot.plugins import steps
+
     f.addStep(steps.Configure())
 
+.. bb:step:: CMake
+
+CMake
++++++
+
+.. py:class:: buildbot.steps.cmake.CMake
+
+This is intended to handle the :command:`cmake` step for projects that use `CMake-based build systems <http://cmake.org>`_.
+
+.. note::
+
+   Links below point to the CMake v3.5 documentation.
+   Please make sure that you check the documentation for the CMake you use.
+
+In addition to the parameters :bb:step:`ShellCommand` supports, this step accepts the following parameters:
+
+``path``
+    Either a path to a source directory to (re-)generate a build system for it in the current working directory.
+    Or an existing build directory to re-generate its build system.
+
+``generator``
+    A build system generator.
+    See `cmake-generators(7) <https://cmake.org/cmake/help/v3.5/manual/cmake-generators.7.html>`_ for available options.
```

Comment:
My mistake. I remember the way of using cmake and wasn't traditional...

I mean, there might be several strategy around cmake:
- indeed, you might want to run the generator in the ci to product the build system files (Makefile, Ninja,...). In the CI, you need to run the generator and then the build system. => for that, don't you want to add a way to run the build system automatically, or you want user to add another "build" step that would run the ninja command?
- in my case, we generated the makefiles once, committed them, and made heavy modifications on it, so we didn't run the generator in the CI... but, yes, it's not the canonic way of working with cmake !